### ISSUE TITLE
Possible to set a seed value?

### ISSUE THREAD

**Issue by Folcon** [mikera/clisk#10]
Would it be possible to add the ability to set or pass an optional seed for the noise generators?

**Comment by mikera** [comment]
Sounds like a sensible enhancement. Happy to accept a PR for this! 

Otherwise I'll add it when I next get round to revamping this part of the code base

**Comment by Folcon** [comment]
First I'm trying to work out if I can use this to generate 2d terrain a la [libtcod heightmaps](http://doryen.eptalys.net/data/libtcod/doc/1.5.2/html2/heightmap_modify.html?c=true&cpp=true&cs=true&py=true&lua=true) >_<...

Been reading your blog so this has been useful for creating a starting point with voronoi

```
(show (v+ voronoi-function (scale 0.2 (rgb-from-hsl [noise 0.8 noise]))))
```

![v voronoi](https://f.cloud.github.com/assets/123055/1884908/aef784d6-79b3-11e3-8c07-6d32583a9531.png)

```
(show (render-lit (v+ vnoise (scale 0.2 (rgb-from-hsl [noise 0.8 noise]))) (v* 6 plasma)))
```

![v vnoise](https://f.cloud.github.com/assets/123055/1884915/f0094680-79b3-11e3-86b1-a48d9550b0a7.png)

or

```
(show (render-lit (v- vnoise (scale 0.2 (rgb-from-hsl [noise 0.8 noise]))) (v* 6 plasma)))
```

![v- vnoise](https://f.cloud.github.com/assets/123055/1884914/e8702fe2-79b3-11e3-966e-fceed2c2ac6a.png)

But still getting to grips with it, changing the way it behaves and how to go about doing things like simulating rain erosion or placing hills and mountains :)...

Kudo's for this though, the lib is pretty fun to use, if I work out how to do some of this, I'll see if I can write up a tutorial or something!

**Comment by mikera** [comment]
Cool, glad you are making some fine creations with it!

I found that multiplying a plasma with some other noisy function (e.g. another plasma with a different scale/offset) is pretty good for heightmaps.

**Comment by mikera** [comment]
Also did you find the heightmap example in `clisk/samples/gallery.clj`?

**Comment by Folcon** [comment]
Actually I didn't thanks! I assume you mean the `:fractal-landscape`?
Also wow, I've also been looking over your other repo's, I'll probably be trying out the `orculje` and looking over `alchemy` in my quest to build my own roguelike :)...

**Comment by mikera** [comment]
Yep that's the one. Should be fairly easy to customise with other terms etc. And you can remove the `render-lit` if you just want the colours.

Have fun with Orculje etc.! I'm still maintaining/patching these from time to time, if you see anything you particularly want to enhance then let me know (or PRs welcome!)
